server.go

```diff
@@ -413,3 +413,8 @@ func (s *Server) Stop() error {
 
 	return nil
 }
+
+// IsRunning returns true if the server is running.
+func (s *Server) IsRunning() bool {
+	return s.started == 1 && s.shutdown == 0
```

Comment:
Should use the `atomic` package to load these.

Reply:
Following your other comment, will recreate the service if necessary rather than restart.